### ISSUE TITLE
NullRef exception when ID is set to null when creating Mapping

### ISSUE THREAD

**Issue by dariusdamalakas** [wiremock/wiremock#1164]
# Bug reports

We're a lot more likely to look at and fix bugs that are clearly described and reproduceable. Please including the following details when reporting a bug:

- [ ] Which version of WireMock you're using

Docker image, https://hub.docker.com/r/rodolpheche/wiremock/
Latest docker image (as of 2019 07 31), not sure how to check version.

- [ ] How you're starting and configuring WireMock, including configuration or CLI the command line

`docker run -it --rm -p 8081:8080 rodolpheche/wiremock`

- [ ] A failing test case that demonstrates the problem

*Steps to reproduce: *
Create new mapping:

```
POST http://localhost:8081/__admin/mappings HTTP/1.1
Host: localhost:8081
Content-Length: 200

{
   "id": null,
  "uuid": null,
  "name": "id1-name",
  "request": {
    "method": "POST",
    "url": "/sop/PUBSES"
  },
  "response": {
    "body": "hello world",
    "status": 200
  }
}
HTTP/1.1 201 Created
Content-Type: application/json
Transfer-Encoding: chunked
Server: Jetty(9.4.z-SNAPSHOT)

{
  "name" : "id1-name",
  "request" : {
    "url" : "/sop/PUBSES",
    "method" : "POST"
  },
  "response" : {
    "status" : 200,
    "body" : "hello world"
  }
}


```

Failing request (Fiddler output)
```
POST http://localhost:8081/sop/PUBSES HTTP/1.1
Host: localhost:8081
Content-Length: 273

 {
 "id": null,
  "uuid": null,
  "name": "id1-name",
  "request": {
    "method": "POST",
    "url": "/sop/PUBSES"
  },
  "response": {
    "body": "hello world test1",
    "status": 200,
    "headers": {
      "Content-Type": "application/json"
    }
  }
}
HTTP/1.1 500 Server Error
Cache-Control: must-revalidate,no-cache,no-store
Content-Type: text/html;charset=iso-8859-1
Content-Length: 3362
Connection: close
Server: Jetty(9.4.z-SNAPSHOT)

<html>
<head>
<meta http-equiv="Content-Type" content="text/html;charset=utf-8"/>
<title>Error 500 Server Error</title>
</head>
<body><h2>HTTP ERROR 500</h2>
<p>Problem accessing /sop/PUBSES. Reason:
<pre>    Server Error</pre></p><h3>Caused by:</h3><pre>java.lang.NullPointerException
	at com.github.tomakehurst.wiremock.http.StubResponseRenderer.renderDirectly(StubResponseRenderer.java:94)
	at com.github.tomakehurst.wiremock.http.StubResponseRenderer.buildResponse(StubResponseRenderer.java:64)
	at com.github.tomakehurst.wiremock.http.StubResponseRenderer.render(StubResponseRenderer.java:56)
	at com.github.tomakehurst.wiremock.http.AbstractRequestHandler.handle(AbstractRequestHandler.java:50)
	at com.github.tomakehurst.wiremock.servlet.WireMockHandlerDispatchingServlet.service(WireMockHandlerDispatchingServlet.java:120)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
	at wiremock.org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:873)
	at wiremock.org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:542)
	at wiremock.org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:255)
	at wiremock.org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1345)
	at wiremock.org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:203)
	at wiremock.org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:480)
	at wiremock.org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:201)
	at wiremock.org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1247)
	at wiremock.org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:144)
	at wiremock.org.eclipse.jetty.server.handler.gzip.GzipHandler.handle(GzipHandler.java:753)
	at wiremock.org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:126)
	at wiremock.org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
	at wiremock.org.eclipse.jetty.server.Server.handle(Server.java:502)
	at wiremock.org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:370)
	at wiremock.org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:267)
	at wiremock.org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:305)
	at wiremock.org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:103)
	at wiremock.org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:117)
	at wiremock.org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:333)
	at wiremock.org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:310)
	at wiremock.org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:168)
	at wiremock.org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:126)
	at wiremock.org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:366)
	at wiremock.org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:765)
	at wiremock.org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:683)
	at java.lang.Thread.run(Thread.java:748)
</pre>
<hr><a href="http://eclipse.org/jetty">Powered by Jetty:// 9.4.z-SNAPSHOT</a><hr/>

</body>
</html>


```

**Comment by wasiek** [comment]
Hi @dariusdamalakas , problem lays in your mapping, you are setting:
`   "id": null,
  "uuid": null,
`
The NullPointerException comes from  `stubMapping.getId().toString()` as you set it to null you cannot call toString().

I would advice you to not send `id` and `uuid` (at least not `null`). I've checked your example without those two properties and it worked.

**Comment by dariusdamalakas** [comment]
Thanks @wasiek,  that was the intention of the payload - to see how wiremock handles that.   I stumbled accidentally by this, was an issue with my code. 

IMO wiremock should return HTTP 400 saying id can not be null, or perhaps even ignore `ID` on `post` methods, because POST creates a new mapping.  ID should be accepted only on PUT (and thats where null should also not be allowed, and should result in 400)

**Comment by ChunjuanLi** [comment]
Hi @tomakehurst  @dariusdamalakas @wasiek  I am trying to fix the issue however I am not able to reproduce the issue now, and it seems wiremock always provides a random ID whenever ID is Null/Empty:
code : https://github.com/wiremock/wiremock/blame/master/src/main/java/com/github/tomakehurst/wiremock/core/WireMockApp.java#L228 

Can you please confirm if this is still a valid bug? Thanks!

**Comment by tomakehurst** [comment]
Looks like this was fixed in 5fac15f7fb527da2db14b0ab5adc4e45ae73b3fa

**Comment by ChunjuanLi** [comment]
@tomakehurst Thanks for confirming!
